### PR TITLE
[SPARK-40009][PYTHON][DOCS] Add missing doc string info to DataFrame API

### PR DESCRIPTION
**Pull request by khalidmammadov**
Adding examples and parameters description to union and unionAll 

### What changes were proposed in this pull request?
Documentation enhancements

### Why are the changes needed?
Help user to understand functions

### Does this PR introduce _any_ user-facing change?
Yes, improves documentation

### How was this patch tested?
unit test

### CODE REVIEW

**Comment by khalidmammadov** [comment]
@HyukjinKwon @srowen please review

**Review by srowen — COMMENTED** [review]
Seems fine; do you want to add more missing docstrings in one go?

**Comment by HyukjinKwon** [comment]
Yeah can we fix the whole file instead of one API alone?

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -2520,6 +2550,38 @@ def unionAll(self, other: "DataFrame") -> "DataFrame":
         (that does deduplication of elements), use this function followed by :func:`distinct`.
 
         Also as standard in SQL, this function resolves columns by position (not by name).
+
+        Note: This is identical to :func:`union` function
```
Comment (HyukjinKwon) [inline python/pyspark/sql/dataframe.py:2554]: Let's just probably mention that it's an alias, and remove Examples section.

**Comment by HyukjinKwon** [comment]
BTW thanks for working on this @khalidmammadov

**Comment by khalidmammadov** [comment]
Sure, will work on whole file and revert

**Comment by khalidmammadov** [comment]
@HyukjinKwon @srowen please review

**Comment by khalidmammadov** [comment]
I have also checked how they look using `bundle exec jekyll serve --host 0.0.0.0`

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**